### PR TITLE
community: Chat Huggingface and HuggingFace TGI llms

### PR DESCRIPTION
**Pull request by minhna1112**
community: 

  - **Description:** 
   - [x] The method `_resolve_model_id()` is only called when `model_id` is not explicitly defined, in order to avoid calling `list_inference_endpoints` when connecting `HuggingFaceTextGenInference` to a self-hosted TGI server.
  - [x]  For  `HuggingFaceTextGenInference` clients that send request to a HuggingFace-deployed Inference Endpoints, a Bearer Token is needed for authorization.
  - [x] More integrations tests is added for `HuggingFaceTextGenInference` 
  - **Issue:** Referring to this current [issue](https://github.com/huggingface/text-generation-inference/issues/747) in HuggingFace TGI repo
  - **Dependencies:** No dependencies for this change
  - **Twitter handle:**  @MinhN2804

### CODE REVIEW

**Comment by minhna1112** [comment]
@hwchase17 Can you please help me merge this PR?

**Review by baskaryan — COMMENTED** [review]
cc @andrewrreed

**Comment by minhna1112** [comment]
@andrewrreed  @baskaryan  I have fixed the spelling error occurred during the last check.

File: libs/community/langchain_community/llms/huggingface_text_gen_inference.py [hunk 1]
```diff
@@ -129,8 +131,17 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
 
     @root_validator()
     def validate_environment(cls, values: Dict) -> Dict:
-        """Validate that python package exists in environment."""
-
+        """Validate that api key and python package exists in environment."""
+        huggingfacehub_api_token = get_from_dict_or_env(
+            values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN"
+        )
```
Comment (baskaryan) [inline libs/community/langchain_community/llms/huggingface_text_gen_inference.py:137]: will raise an error if no default is provided and token isn't provided, can we do
```suggestion
        huggingfacehub_api_token = get_from_dict_or_env(
            values, "huggingfacehub_api_token", "HUGGINGFACEHUB_API_TOKEN", default=""
        )
```

**Comment by minhna1112** [comment]
@baskaryan  I have pushed [0f9784c](https://github.com/langchain-ai/langchain/pull/15582/commits/0f9784c94309953da4c88000f5a7493494b41b6e) to resolve library diffs during CI. Just tell me if there are more things need to be changed. Thanks a lot!

**Comment by minhna1112** [comment]
@baskaryan  Can you please help me merge and close this PR? Thanks a lot!

**Comment by efriis** [comment]
closing because this is against the deprecated community huggingface integrations. Feel free to resubmit against the partner package (libs/partners/huggingface) if it's still missing there!

**Pull request closed without merging**